mobile/lib/widgets/settings/networking_settings/external_network_preference.dart

```diff
@@ -0,0 +1,199 @@
+import 'package:immich_mobile/extensions/build_context_extensions.dart';
+import 'package:immich_mobile/models/auth/auxilary_endpoint.model.dart';
+import 'package:immich_mobile/entities/store.entity.dart' as db_store;
+import 'package:immich_mobile/widgets/settings/networking_settings/endpoint_input.dart';
+
+class ExternalNetworkPreference extends HookConsumerWidget {
+  const ExternalNetworkPreference({super.key, required this.enabled});
+
+  final bool enabled;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final entries =
+        useState([AuxilaryEndpoint(url: '', status: AuxCheckStatus.unknown)]);
+    final canSave =
+        useState(entries.value.every((e) => e.status == AuxCheckStatus.valid));
+
+    saveEndpointList() {
+      canSave.value =
+          entries.value.every((e) => e.status == AuxCheckStatus.valid);
+
+      final endpointList = entries.value
+          .where((url) => url.status == AuxCheckStatus.valid)
+          .toList();
+
+      final jsonString = jsonEncode(endpointList);
+
+      db_store.Store.put(
+        db_store.StoreKey.externalEndpointList,
+        jsonString,
+      );
+    }
+
+    updateValidationStatus(String url, int index, AuxCheckStatus status) {
+      entries.value[index] =
+          entries.value[index].copyWith(url: url, status: status);
+
+      saveEndpointList();
+    }
+
+    handleReorder(int oldIndex, int newIndex) {
+      if (oldIndex < newIndex) {
+        newIndex -= 1;
+      }
+
+      final entry = entries.value.removeAt(oldIndex);
+      entries.value.insert(newIndex, entry);
+      entries.value = [...entries.value];
+
+      saveEndpointList();
+    }
+
+    handleDismiss(int index) {
+      entries.value = [...entries.value..removeAt(index)];
+
+      saveEndpointList();
+    }
+
+    Widget proxyDecorator(
+      Widget child,
+      int index,
+      Animation<double> animation,
+    ) {
+      return AnimatedBuilder(
+        animation: animation,
+        builder: (BuildContext context, Widget? child) {
+          return Material(
+            color: context.colorScheme.surfaceContainerHighest,
+            shadowColor: context.colorScheme.primary.withOpacity(0.2),
+            child: child,
+          );
+        },
+        child: child,
+      );
+    }
+
+    useEffect(
+      () {
+        final jsonString =
+            db_store.Store.tryGet(db_store.StoreKey.externalEndpointList);
+
+        if (jsonString == null) {
+          return null;
+        }
+
+        final List<dynamic> jsonList = jsonDecode(jsonString);
+        entries.value =
+            jsonList.map((e) => AuxilaryEndpoint.fromJson(e)).toList();
+        return null;
+      },
+      [],
+    );
+
+    return Padding(
+      padding: const EdgeInsets.symmetric(horizontal: 16.0),
+      child: Container(
+        clipBehavior: Clip.antiAlias,
+        decoration: BoxDecoration(
+          borderRadius: BorderRadius.circular(16),
```

Comment:
`const BorderRadius.all(Radius.circular(16))`

Reply:
Can't, it is not allowed there 
<img width="1024" alt="image" src="https://github.com/user-attachments/assets/d7f70c0d-ab58-4be2-b989-8cd85518cae4">

Reply:
We should be using the all constructor instead of the circular constructor